ln-gateway/src/bin/ln_gateway.rs

```diff
@@ -51,9 +41,20 @@ async fn generate_config(workdir: &Path, ln_client: &mut ClnRpc) {
     };
     let node_pub_key = secp256k1::PublicKey::from_slice(&node_pub_key_bytes.to_vec()).unwrap();
 
+    // Write gateway config
+    let gateway_cfg = GatewayClientConfig {
+        client_config: federation_client_cfg.clone(),
+        redeem_key: kp_fed,
+        timelock_delta: 10,
+        node_pub_key,
+        api: String::from("http://127.0.0.1:8080"), // FIXME: don't hard-code this
+    };
+    let gw_cfg_file_path: PathBuf = workdir.join("gateway.json");
+    let gw_cfg_file = std::fs::File::create(gw_cfg_file_path).expect("Could not create cfg file");
+    serde_json::to_writer_pretty(gw_cfg_file, &gateway_cfg).unwrap();
+
     // Write user config
     let client_cfg = UserClientConfig(federation_client_cfg);
-
     let client_cfg_file_path: PathBuf = workdir.join("client.json");
     let client_cfg_file =
         std::fs::File::create(client_cfg_file_path).expect("Could not create cfg file");
```

Comment:
Is this still necessary with the gateway registration code?

Reply:
I think we can just `cp` the `federation_client.json` to `client.json` right? Or do we need to make the newtype transparent for serde for that to work? (in which case I'd propose to do that)

Reply:
I got rid of `federation_client.json`. Now we just have `client.json` which can be parsed as `UserClientConfig` or `ClientConfig`.